src/components/BetForm.vue

```diff
@@ -1,12 +1,122 @@
 <template>
-  <form>
-    <h2>Bet Form</h2>
-  </form>
+  <v-form ref="form" v-model="valid">
+    <v-overlay :value="confirmSubmission">
+      <v-card
+        light
+        max-width="400"
+        class="mx-auto"
+      >
+        <v-card-text>
+          <p class="title" v-if="ticker">
+            Confirm bet of {{ amount }} to {{ ticker.name }}?
+          </p>
+        </v-card-text>
+
+        <v-card-actions class="justify-center">
+          <v-btn color="primary" class="mr-4" @click="confirmSubmit">
+            Place bet
+          </v-btn>
+
+          <v-btn class="mr-4" @click="cancelSubmit">
+            Cancel
+          </v-btn>
+        </v-card-actions>
+      </v-card>
+    </v-overlay>
+
+    <div>
+      <v-text-field
+        v-model="amount"
+        :rules="amountRules"
+        label="Amount"
+        type="number"
+        prepend-icon="mdi-currency-usd"
+        required
+      ></v-text-field>
+      <v-autocomplete
+        v-model="ticker"
+        :items="tickers"
+        :rules="tickerRules"
+        item-text="name"
+        item-value="symbol"
+        label="Ticker"
+        return-object
+        required
+      >
+        <template v-slot:selection="data">
+          {{ formatTicker(data.item) }}
+        </template>
+        <template v-slot:item="data">
+          {{ formatTicker(data.item) }}
+        </template>
+      </v-autocomplete>
+
+      <v-btn :disabled="!valid || submitting" class="mr-4" @click="submit">
+        Bet
+      </v-btn>
+    </div>
+  </v-form>
 </template>
 
 <script>
+import tickers from '../data/tickers'
+import * as constants from '../constants'
+
 export default {
-  name: 'betForm'
+  data: () => {
+    const data = {
+      show: false,
+      confirmSubmission: false,
+      amountRules: [minimumAmount],
+      tickerRules: [nonEmptyTicker],
+      tickers
+    }
+
+    resetData(data)
+
+    return data
+  },
+  methods: {
+    formatTicker (ticker) {
+      return `${ticker.name} - ${ticker.symbol}`
+    },
+
+    cancelSubmit () {
+      this.confirmSubmission = false
+    },
+
+    confirmSubmit () {
+      this.show = true
+      this.$store.dispatch('bet', {
+        amount: this.amount,
+        ticker: this.ticker.nameid
+      })
+
+      this.$refs.form.reset()
+      resetData(this)
+    },
+
+    submit (event) {
+      this.confirmSubmission = true
+    }
+  }
+}
+
+function minimumAmount (amount) {
```

Comment:
Can you move this functions to methods?

Reply:
Yes, but why don't you like it the way it is? You can think of these as utility functions that don't need the context of an object to be invoked.